src/ucs/datastruct/arbiter.c

```diff
@@ -46,23 +46,28 @@ static inline void ucs_arbiter_group_head_reset(ucs_arbiter_elem_t *head)
     head->list.next = NULL; /* Not scheduled yet */
 }
 
+static inline void ucs_arbiter_elem_set_scheduled(ucs_arbiter_elem_t *elem,
+                                                  ucs_arbiter_group_t *group)
+{
+    elem->group = group;
+}
+
 void ucs_arbiter_group_push_elem_always(ucs_arbiter_group_t *group,
                                         ucs_arbiter_elem_t *elem)
 {
     ucs_arbiter_elem_t *tail = group->tail;
 
-    UCS_ARBITER_GROUP_GUARD_CHECK(group);
```

Comment:
why is guard removed here?

Reply:
because now we allow recursive push during dispatch (which was the goal of the PR)